src/Core/Managed/Net46/Extensibility/Implementation/RichPayloadEventSource.cs

```diff
@@ -0,0 +1,268 @@
+﻿namespace Microsoft.ApplicationInsights.Extensibility.Implementation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.Tracing;
+    using System.Globalization;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.External;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+
+    internal sealed class RichPayloadEventSource : IDisposable
+    {
+        /// <summary>RichPayloadEventSource instance.</summary>
+        public static readonly RichPayloadEventSource Log = new RichPayloadEventSource();
+        
+        /// <summary>Event source.</summary>
+        internal readonly EventSource eventSource;
+
+        /// <summary>Event provider name.</summary>
+        private const string EventProviderName = "Microsoft-ApplicationInsights";
+
+        /// <summary>
+        /// Initializes a new instance of the RichPayloadEventSource class.
+        /// </summary>
+        public RichPayloadEventSource()
+        {
+            this.eventSource = new EventSource(
+               EventProviderName,
+               EventSourceSettings.EtwSelfDescribingEventFormat,
+               "ETW_GROUP",
+               "{E3AFA6F4-0FEA-4757-AFFF-80B5CA685BB8}");
```

Comment:
where is this constant from? Consider moving it to https://github.com/Microsoft/ApplicationInsights-dotnet/blob/271edd5fd8b9c7d84a9bd72448ec14f9b2145259/src/Core/Managed/Shared/Constants.cs

Also I'm not sure that we need to specify ETW_GROUP - I vaguely remember it was used for UTC. @izikl, can you please advice here?